### ISSUE TITLE
Longer encoding names don't fit in the button

### ISSUE THREAD

**Issue by rdbende** [Akuli/porcupine#1070]
![image](https://user-images.githubusercontent.com/77941087/168466824-5d6e02a6-b98d-40ca-aac1-be7d7de6f708.png)
Longer encoding names (e.g. `shift-jisx0213`, but even `utf-16-be`) don't fit in the encoding button.

**Comment by Moosems** [comment]
Where is the code for the button?

**Comment by Akuli** [comment]
The entire status bar is `porcupine/plugins/statusbar.py`.

**Comment by rdbende** [comment]
@Akuli You're 40 seconds faster 😄

**Comment by Moosems** [comment]
@rdbende @Akuli Quick question for a personal project (unrelated): For the indentation if the line ends with `:` how does porcupine check if the line is commented?

**Comment by Akuli** [comment]
Dumbly. You can see it if you try to parse a text file that contains comments: `if line.startswith("#"):` doesn't get the indentation after it, because Porcupine sees it as `if line.startswith("` followed by a comment. This comes up every once in a while when I use porcupine, but not really often enough to be annoying.

A better solution would be to rely on tokenizing done by the syntax highlighter. If something is highlighted with the colors that strings use, then any `#` characters inside it could be skipped. Of course, this would mean that your indentations get weird if you turn off syntax highlighting, but nobody does that anyway :)

**Comment by Akuli** [comment]
And for multiple languages: the comment character is set in `filetypes.toml`. Here are all default values that Porcupine uses for various languages:

```
$ grep 'comment_prefix =' porcupine/default_filetypes.toml 
comment_prefix = '#'
comment_prefix = '#'
comment_prefix = '//'
comment_prefix = '//'
comment_prefix = '//'
comment_prefix = '//'
comment_prefix = '//'
comment_prefix = '#'
comment_prefix = '#'
comment_prefix = '#'
comment_prefix = '#'
comment_prefix = '#'
comment_prefix = '%'
comment_prefix = ';'
```

**Comment by Moosems** [comment]
Does it check for multi line comments?

**Comment by Moosems** [comment]
> Dumbly. You can see it if you try to parse a text file that contains comments: `if line.startswith("#"):` doesn't get the indentation after it, because Porcupine sees it as `if line.startswith("` followed by a comment. This comes up every once in a while when I use porcupine, but not really often enough to be annoying.

You could use `if (line.strip()).startswith("#"):` to make it work with space indentation and make sure to remove tab indentation before

**Comment by Akuli** [comment]
It doesn't work with multiline comments. It's tempting to say that it's a feature, not a bug: I often write a small snippet of Python code inside a `"""` docstring.

And yes, I usually do the `.startswith()` on a stripped line when I need to parse something from a text file :)

**Comment by Moosems** [comment]
FIxed in #1090, can I get a review?
